src/Upgrade/Application/Strategy/Common/PackagePostUpdateHandler/PropelUpdateHandler.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace Upgrade\Application\Strategy\Common\PackagePostUpdateHandler;
+
+use Core\Infrastructure\Service\ProcessRunnerServiceInterface;
+use Upgrade\Application\Dto\StepsResponseDto;
+
+class PropelUpdateHandler implements HandlerInterface
+{
+    /**
+     * @var string
+     */
+    protected const PROPEL_PACKAGE_NAME = 'propel/propel';
+
+    /**
+     * @var string
+     */
+    protected const PROJECT_CONSOLE_PATH = 'vendor/bin/console';
+
+    /**
+     * @var string
+     */
+    protected const TRANSFER_GENERATE_COMMAND = 'transfer:generate';
+
+    /**
+     * @var string
+     */
+    protected const PROPEL_SCHEMA_COPY_COMMAND = 'propel:schema:copy';
+
+    /**
+     * @var string
+     */
+    protected const PROPEL_MODEL_BUILD_COMMAND = 'propel:model:build';
+
+    /**
+     * @var array<string>
+     */
+    protected const GENERATED_MODEL_PATH_LIST = [
+        'data/cache/propel',
+        'src/Orm/Zed/**/Persistence/Base/',
+        'src/Orm/Zed/**/Persistence/Map/',
+        'src/Generated/',
+        'src/Orm/Propel/*',
+    ];
+
+    /**
+     * @var \Core\Infrastructure\Service\ProcessRunnerServiceInterface
+     */
+    protected ProcessRunnerServiceInterface $processRunner;
+
+    /**
+     * @param \Core\Infrastructure\Service\ProcessRunnerServiceInterface $processRunner
+     */
+    public function __construct(ProcessRunnerServiceInterface $processRunner)
+    {
+        $this->processRunner = $processRunner;
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsExecutionDto
+     *
+     * @return bool
+     */
+    public function isApplicable(StepsResponseDto $stepsExecutionDto): bool
+    {
+        $composerLockDiffDto = $stepsExecutionDto->getComposerLockDiff();
+        if (!$composerLockDiffDto) {
+            return false;
+        }
+
+        return $this->hasPropel($composerLockDiffDto->getRequireChanges());
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsExecutionDto
+     *
+     * @return \Upgrade\Application\Dto\StepsResponseDto
+     */
+    public function handle(StepsResponseDto $stepsExecutionDto): StepsResponseDto
+    {
+        $commandList = [
+            sprintf('rm -rf %s', implode(' ', self::GENERATED_MODEL_PATH_LIST)),
```

Comment:
Can we replace `rm -rf` with this one https://symfony.com/doc/current/components/filesystem.html#remove?
It is more stable and we can control failures in this case.

Reply:
++